controller/src/controller.rs

```diff
@@ -261,6 +272,28 @@ fn sort_shadows(shadows: &mut Vec<BottlerocketShadow>, associated_brs_name: &str
     }
 }
 
+/// Fetch the environmental variable to determine the max concurrent update nodes number.
+fn get_max_concurrent_update() -> Result<usize> {
+    let max_concurrent_update = env::var(MAX_CONCURRENT_UPDATE_ENV_VAR)
+        .context(controllerclient_error::EnvVarNotExist {
+            variable: MAX_CONCURRENT_UPDATE_ENV_VAR.to_string(),
+        })?
+        .to_lowercase();
+
+    if max_concurrent_update.eq("unlimited") {
+        return Ok(usize::MAX);
+    } else {
+        max_concurrent_update
+            .parse::<usize>()
+            .context(controllerclient_error::MaxConcurrentUpdateParseError)
+    }
+}
+
+/// Determine if a BottlerocketShadow is in default or None status.
+fn is_initial_state(brs: &BottlerocketShadow) -> bool {
+    brs.status == None
+        || brs.status.clone().unwrap().current_state == BottlerocketShadowState::default()
```

Comment:
It's better to avoid using `unwrap` here

Reply:
It is safe to use `unwrap()` here because the condition is `or` condition. The program will only reach to `brs.status.clone().unwrap()` if `brs.status != None`.